example_caller/README.md

```diff
@@ -1,62 +1,165 @@
-# `processor_caller`
+Below is the updated documentation with a fourth requirement added for the Orchestrator. This new section explains that the Orchestrator must be running with the write canister ID and the JOB_SCHEDULE set to run every 5 seconds for the best results.
 
-Welcome to your new `processor_caller` project and to the Internet Computer development community. By default, creating a new project adds this README and some template files to your project directory. You can edit these template files to customize your project and to include your own code to speed up the development cycle.
+---
 
-To get started, you might want to explore the project directory structure and the default configuration file. Working with this project in your development environment will not affect any production deployment or identity tokens.
+# Benchmarking Setup
 
-To learn more before you start working with `processor_caller`, see the following documentation available online:
+This project can be used to benchmark requests. Before you begin testing, you must ensure that three Internet Computer (IC) canisters, a prover server, and an Orchestrator are up and running.
+
+## PROVER Server
+
+- **Default URL:**  
+  Your local verity prover will be used by default at `http://localhost:8080`.  
+- **Custom URL:**  
+  Change this by setting the environment variable:  
+  ```bash
+  export PROVER_URL="https://your-prover-url"
+  ```
+
+## Required Canisters & Orchestrator
+
+The following must be deployed and running in order:
+
+1. **Managed Verifier**  
+   Repository: [Managed Verifier](https://github.com/usherlabs/verity-dp/tree/main/ic/managed/verifier)  
+   *Note:* The canister ID for the Managed Verifier will be required in the next step.
+
+2. **ADC Processor**  
+   Location: [ADC Processor](../processor/ic)  
+   To configure this canister, set the verifier address by calling the `set_verifier_canister` method.
+
+3. **ADC Caller Example**  
+   Run the ADC_Caller example and set the ADC address using the `set_adc_address` method.
+
+4. **Orchestrator**  
+   The Orchestrator should be running from the `./orchestrator` directory. Make sure to launch it with the following requirements:
+   - **Write Canister ID:** Provide the write canister ID as required.
+   - **Job Schedule:** Set the environment variable `JOB_SCHEDULE` to `*/5 * * * * *` to schedule the job to run every 5 seconds for optimal benchmarking results.
+   
+   For example, you might start the orchestrator as follows:
+   ```bash
+   export JOB_SCHEDULE="*/5 * * * * *"
+   ./orchestrator --write-canister-id <your_write_canister_id>
+   ```
+
+## Running the Benchmark
+
+Once all three canisters and the orchestrator are running, you can edit the benchmark URL (located in `./__tests__/test_adc.test.ts`) if needed. Then, prepare and run your tests using either npm or yarn:
+
+```bash
+# Using npm:
+npm run prep
+npm run test
+
+# Or using yarn:
+yarn prep
+yarn test
+```
```

Comment:
The instruction doesn't look like it's going to work. The command `yarn prep` restarts `dfx` with instruction `dfx stop; dfx start --background --clean;` will wipe out any preparation steps to deploy required canisters.

There is no `package-lock.json` file but `yarn.lock` is present, which tells us this is a project managed by `yarn`. Please remove the option to run with `npm`.

It could be good to have `yarn install` command before `prep` and `test`.